### PR TITLE
Capture

### PR DESCRIPTION
**Pull request by albertodvp**
## Summary by CodeRabbit

* **New Features**
  * Event-stream processing with automatic error detection, snapshot capture, and resource-safe file streaming
  * New validators and options for collection and capture configuration (polling, windows, min-context)

* **Bug Fixes**
  * Improved IO error reporting (file-not-found, permission denied) and guaranteed file-handle cleanup

* **Tests**
  * Added comprehensive tests for capture streams, window management, types, and serialization

* **Documentation**
  * Quick-start install step removed; note on post-window event limits added

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

### CODE REVIEW

File: hs-spanshot/src/Capture.hs [hunk 1]
```diff
@@ -60,3 +73,139 @@ consider optimizing with:
...
+minContextEvents = 2
+Result: [t=6, t=7, t=8, t=10]  -- Cutoff at t=5
+
+-- Sparse logging (fallback to count-based)
+buffer = [t=0, t=1, t=2]
+newEvent = t=100
+preWindowDuration = 5s
+minContextEvents = 3
+Result: [t=1, t=2, t=100]  -- Keep last 3
+@
```
Comment (albertodvp) [inline hs-spanshot/src/Capture.hs]: Do we have tests for these?

File: hs-spanshot/test/CaptureStreamSpec.hs [hunk 2]
```diff
@@ -0,0 +1,160 @@
+{-# LANGUAGE OverloadedStrings #-}
```
Comment (albertodvp) [inline hs-spanshot/test/CaptureStreamSpec.hs]: This file should also test that the boundaries (e.g too early events should not be consider and too late events)? Is it covered somewhere else?
